<launch>
  <arg name="map_name"  default="map_gf1"/>
  <arg name="map_file"  default="$(find tb3_swarm)/maps/$(arg map_name).yaml"/>
  <arg name="open_rviz" default="true"/>
  <arg name="robot1_ns" default="robot1"/>
  <arg name="robot2_ns" default="robot2"/>

  <!-- Run the map server -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
    <param name="frame_id"  value="/map"/>
  </node>

  <!-- Rviz -->
  <group if="$(arg open_rviz)">
    <node pkg="rviz" type="rviz" name="rviz" required="true"
        args="-d $(find tb3_swarm)/rviz/navigation/multirobot_navigation.rviz"/>
  </group>

  <!-- ROBOT1 -->
  <group ns="robot1">
    <arg name="_robot_name"  value="robot1" />
    <arg name="_tf_robot" value="$(arg _robot_name)_tf" />
    <param name="tf_prefix" type="string" value="$(arg _tf_robot)"/>

    <include file="$(find tb3_swarm)/launch/navigation/multi_amcl.launch">
      <arg name="robot_namespace" value="$(arg robot1_ns)"/>
      <arg name="initial_pose_x"  value="0.0"/>
      <arg name="initial_pose_y"  value="0.0"/>
      <arg name="initial_pose_a"  value="0.0"/>
    </include>

    <include file="$(find tb3_swarm)/launch/navigation/multi_move_base.launch">
      <arg name="robot_namespace" value="$(arg robot1_ns)"/>
    </include>
  </group>

  <!-- ROBOT1 -->
  <group ns="robot2">
    <arg name="_robot_name"  value="robot2" />
    <arg name="_tf_robot" value="$(arg _robot_name)_tf" />
    <param name="tf_prefix" type="string" value="$(arg _tf_robot)"/>

    <include file="$(find tb3_swarm)/launch/navigation/multi_amcl.launch">
      <arg name="robot_namespace" value="$(arg robot2_ns)"/>
      <arg name="initial_pose_x"  value="0.0"/>
      <arg name="initial_pose_y"  value="0.0"/>
      <arg name="initial_pose_a"  value="0.0"/>
    </include>

    <include file="$(find tb3_swarm)/launch/navigation/multi_move_base.launch">
      <arg name="robot_namespace" value="$(arg robot2_ns)"/>
    </include>
  </group>

  <!-- Other Functionalities -->

  <!-- Like Task Assignment -->


</launch>
